src/platforms/hosted/dap_swd.c

```diff
@@ -236,7 +280,7 @@ static uint32_t dap_swd_clear_error(adiv5_debug_port_s *const target_dp, const b
 		 * into the expected state.
 		 */
 		dap_line_reset();
-		if (target_dp->version >= 2)
+		if (dap_has_swd_sequence && target_dp->version >= 2)
```

Comment:
This cannot be guarded like this. Failure to execute the TARGETSEL phase on a DPv2 target *is an error*. Please fix dap_write_reg_no_check() instead.

Reply:
It is only an error if TARGETSEL is changed from 0 to anything other than 0. It is set up to 0 already with the line reset. For an adapter < V1.2 we never get to a target not 0.

Reply:
BMDA always sets TARGETSEL to a non-zero value for DPv2 targets, and this isn't dependant on the adaptor. It's dependant on the target. TARGETSEL is also non-zero out of reset for DPv2 targets but must also be set to the selector for the DP we want to talk to, otherwise this will incorrectly switch DPs mid communication when doing error recovery. It's straight-up incorrect to guard this like you have. Please fix `dap_write_reg_no_check()` to work with < v1.2 adaptors otherwise this logic will be incorrect and do weird (bad) things.